### ISSUE TITLE
When not managing IAM role, don't require permission to pass role

### ISSUE THREAD

**Issue by kislyuk** [aws/chalice#479]
We use a continuous deployment role with limited privileges. In particular, the deployment role has read-only IAM permissions. When first deploying the function, we use a privileged account that sets up the Lambda IAM role and policy. (We turn off policy autogeneration.)

Up to and including Chalice 0.9, the deploy command correctly did not try to write anything to IAM. However, starting with Chalice 0.10, the call to update_function_configuration unconditionally tries to update the Lambda IAM role ARN, even if it has not changed. This results in an error of this form:

```
 An error occurred (AccessDeniedException) when calling the 
 UpdateFunctionConfiguration operation: User: 
 arn:aws:iam::ACCT:user/CI-USER is not authorized to perform: 
 iam:PassRole on resource: arn:aws:iam::ACCT:role/lambda-dev
```
Obviously, allowing iam:PassRole would defeat the whole purpose/premise of restricting privileges of the CI/CD account.

I propose that the update_function_configuration step compute a diff of what has actually changed, and not try to modify Lambda settings that have not changed.

**Comment by jamesls** [comment]
Thanks for reporting.  Interesting, I would have expected a similar diff to be computer server side and not require iam:PassRole if the role doesn't change.  We can try to get this fixed in chalice.

Out of curiosity, don't you need still need iam:PassRole when you first create the function?  Even if we get the update_function_configuration updated to compute a diff, I'm not sure how we can solve this for the initial create_function case.

**Comment by kislyuk** [comment]
Absolutely, I need iam:PassRole when I first create the function. The way we do this is first create the function from a privileged account, and then let CI/CD deploy updates to it when tests pass.

**Comment by jamesls** [comment]
Ok makes sense, we can get this updated.

**Comment by kislyuk** [comment]
Just a gentle ping to see if there is any possible movement on this :)
